### DISCUSSION TITLE
Solidity vs Web3.py

### DISCUSSION THREAD

**Discussion by Verde888** [smartcontractkit/full-blockchain-solidity-course-py#338] [General]
Hi everyone!
Maybe this question is a bit naive, but I've been through almost 5h of the course and I keep asking myself, "Are we going to learn some more Solidity here"?
Are pure Solidity and its Python (or JS) implementations alternatives to each other, or do you need to learn both to do anything real in this space? Is it always going to look like, we build a contract in Solidity and then play around it in Python?
According to the data I've seen, most developers use the Remix IDE, which means they use pure Solidity.... Of course, I tried researching this,  but maybe someone could explain it to me in 3 sentences?

Good luck with the course!

**Comment by cromewar** [comment]
Hello @Verde888 it depends a lot on what you want to accomplish, as you know Solidity is for smart contract development, you can 100% use just Solidity with remix for your contracts, take in consideration even remix uses javascript web3 for the deployments and  interaction, this allows you to develop and test fast but let's say those contracts are not "user friendly" as you force the users to interact direct on EtherScan for example.

So, if you want to develop a large and scalable project (special when working on a  team) you would like to set up a local environment, and that's when you should use Pyhton or JavasScript to get access to Web3, of course we use frameworks like Truffle (Js) and Brownie (py) because they handle Web3 stuff automatically. Also if you want to integrate a front-end for your smart contract you would install all the dependencies and libraries on your local environment, not mentioning versioning and a lot of other stuff.

So.. the answer is: if you are going to work on small projects and contracts, remix and solidity are just fine. If you want to create a bigger scalable project, you should develop on a local environment using Javascript or Python, being both equally good.

**Reply by Verde888** [reply]
Thank you for your explanation! I am sure it will help many people like me that are still a bit confused with all those technologies :)
